### PR TITLE
Fix CO-255 retained claim admission capacity

### PR DESCRIPTION
**Pull request by Kbediako**
## What

Fix CO-255 retained-claim admission accounting so stale retained handoff residue no longer flattens control-host admission after relaunch, while queued provider workers still consume host-global capacity.

## Why

Post-relaunch control-host state could stay flat with two real running provider issues plus retained residue. `co-status` reported capacity available, but retained-claim reconciliation and direct admission did not draw the same line between non-occupying terminal/released residue and queued/live provider workers.

This stays separate from CO-248's plain `released/not_active` blocked-cache reclaim seam, CO-41 refresh lifecycle stalls, and stale top-level tracked fallback issues such as CO-223.

## How

- Counts `in_progress` and `queued` provider workers against host-global capacity in both poll refresh and direct admission.
- Uses retained queued-claim issue state for per-state admission caps only when the queued retained state is non-terminal and admissible.
- Treats stale queued terminal claim state as unknown for direct admission and fresh-discovery state-slot reporting.
- Skips released queued drains only when the released claim matches the queued run identity, so unrelated host-global queued workers still consume capacity.
- Resolves queued run identity before synthetic claim keys so manifestless in-flight retained claims are not double-counted.
- Adds CO-255-shaped, queued-run, terminal-state, and released-claim identity regression coverage.

## Validation

- Current head: `ebe0329327d3be1c5640a9a5ba7dd3c72cc17855`
- `node scripts/delegation-guard.mjs`
- Docs-review child stream: `.runs/linear-b1835ea5-3d28-493a-934e-1b38048270ce-co255-docs-review/cli/2026-04-20T02-57-50-152Z-2fc42245/manifest.json`
- `npx vitest run orchestrator/tests/ProviderIssueHandoff.test.ts -t "released queued drains|different run|foreign queued worker|foreign queued retained|queued terminal|fresh-discovery state slots"` (5 passed, 334 skipped)
- `npx vitest run --config vitest.config.core.ts orchestrator/tests/ProviderIssueHandoff.test.ts --testNamePattern 'queued terminal|fresh-discovery state slots|foreign active run'` (2 passed, 335 skipped)
- `git diff --check`
- `npm run build`
- `npm run lint` (passes with 3 pre-existing warnings in `orchestrator/tests/DelegationMcpHealth.test.ts`)
- `npm run test` (346 files, 4423 tests)
- `node scripts/spec-guard.mjs --dry-run`
- `npm run docs:check`
- `npm run docs:freshness` (4289 docs, 4292 registry entries)
- `npm run repo:stewardship` (5393 tracked files, 0 action-required)
- `DIFF_BUDGET_OVERRIDE_REASON="CO-255 intentionally keeps the docs-first packet and provider admission regressions in one PR so reviewers can validate the retained/queued capacity boundary together; nonessential spec wording was trimmed, but stacked coverage remains over the default line cap after adding the queued terminal and released-claim identity-mismatch regression coverage." node scripts/diff-budget.mjs --base origin/main`
- `npm run pack:smoke`
- Manifest-backed `npm run review -- --task linear-b1835ea5-3d28-493a-934e-1b38048270ce --commit fd3c6d0b2`: `status=succeeded`, `review_outcome=bounded-success`
- Final checklist-only review retry: latest telemetry `../../.runs/linear-b1835ea5-3d28-493a-934e-1b38048270ce/cli/2026-04-20T02-51-28-865Z-7ca0f6d6/review/telemetry.json` records `status=failed`, `review_outcome=failed-boundary`, `termination_boundary.kind=startup-anchor`
- Manual fallback review/elegance: `out/linear-b1835ea5-3d28-493a-934e-1b38048270ce/manual/20260420T0621Z-review-fallback.md` found no blockers for the final checklist-only follow-up
- `git fetch origin refs/heads/main:refs/remotes/origin/main && git merge-base --is-ancestor origin/main HEAD` (`origin-main-is-ancestor`)
- `codex-orchestrator pr ready-review --pr 567 --quiet-minutes 10` completed cleanly at 2026-04-20T06:43:59Z

Diff budget override: CO-255 intentionally keeps the docs-first packet and provider admission regressions in one PR so reviewers can validate the retained/queued capacity boundary together; nonessential spec wording was trimmed, but stacked coverage remains over the default line cap after adding the queued terminal and released-claim identity-mismatch regression coverage.

Linear: CO-255

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved admission and capacity handling so queued provider runs and retained claims no longer incorrectly block admitting newly Ready issues after rebuilds/restarts.

* **Tests**
  * Expanded test coverage for admission, capacity, queued/retained and edge-case drain scenarios to prevent regressions.

* **Documentation**
  * Added PRD, tech spec, action plan, task checklist and registry entries with validation steps, rollback guidance and ownership details.

### CODE REVIEW

File: .agent/task/linear-b1835ea5-3d28-493a-934e-1b38048270ce.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- [ ] `npm run docs:check`. Evidence: pending.
+- [ ] `npm run docs:freshness`. Evidence: pending.
+- [ ] `npm run repo:stewardship`. Evidence: pending.
+- [ ] `node scripts/diff-budget.mjs`. Evidence: pending.
+- [ ] Manifest-backed standalone review and explicit elegance/minimality pass before review handoff. Evidence: pending.
+
+## Handoff
+- [ ] Parent refreshes the single workpad with docs-review, implementation, validation, and review/elegance status before handoff.
+- [ ] Parent attaches a PR and completes the normal review-state prerequisites before `In Review`.
+
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply .agent/task/linear-b1835ea5-3d28-493a-934e-1b38048270ce.md]: @coderabbitai Fixed in 207155057: cleared pending validation placeholders in both checklist mirrors with concrete local validation evidence, and also applied the PRD table spacing nit. Please re-review/resolve.

File: .agent/task/linear-b1835ea5-3d28-493a-934e-1b38048270ce.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+- TECH_SPEC mirror: `docs/TECH_SPEC-linear-b1835ea5-3d28-493a-934e-1b38048270ce.md`
+- ACTION_PLAN: `docs/ACTION_PLAN-linear-b1835ea5-3d28-493a-934e-1b38048270ce.md`
+- Shared source 0 anchor: `ctx:sha256:de1e154c0eeab0ff773e815641b72013b7de2f3a9f39ddc170e520646a5c300a#chunk:c000001`
+
+## Docs-First
+- [x] PRD drafted for the stale tracked plus retained active/handoff_failed admission bug. Evidence: `docs/PRD-linear-b1835ea5-3d28-493a-934e-1b38048270ce.md`.
+- [x] TECH_SPEC drafted with issue-family boundaries, protected terms, parity matrix, and validation plan. Evidence: `tasks/specs/linear-b1835ea5-3d28-493a-934e-1b38048270ce.md`.
+- [x] ACTION_PLAN drafted for diagnosis, narrow implementation, focused regressions, and validation/review handoff. Evidence: `docs/ACTION_PLAN-linear-b1835ea5-3d28-493a-934e-1b38048270ce.md`.
+- [x] Pre-implementation issue-quality review recorded in the spec readiness gate. Evidence: `tasks/specs/linear-b1835ea5-3d28-493a-934e-1b38048270ce.md`.
+- [x] Registry and mirrors updated: `tasks/index.json`, `docs/TASKS.md`, `docs/docs-freshness-registry.json`, docs TECH_SPEC mirror, and `.agent/task` mirror. Evidence: those files.
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply .agent/task/linear-b1835ea5-3d28-493a-934e-1b38048270ce.md]: @coderabbitai Fixed in 207155057: replaced the vague evidence wording with explicit task, docs, registry, TECH_SPEC mirror, and .agent/task file paths. Please re-review/resolve.

File: orchestrator/src/cli/control/providerIssueHandoff.ts [hunk 3]
```diff
@@ -6269,6 +6299,19 @@ function shouldProviderClaimOccupyPollDispatchSlot(
...
+  claim: Pick<ProviderIntakeClaimRecord, 'state' | 'retry_queued'>,
+  run: Pick<ProviderIssueRunRecord, 'status'> | null
+): boolean {
+  if (run?.status === 'in_progress' || run?.status === 'queued') {
+    return true;
+  }
+  if (claim.retry_queued === true) {
+    return true;
+  }
+  return shouldProviderClaimOccupyPollDispatchSlot(claim);
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/providerIssueHandoff.ts:6373]: Fixed in b3c8c3379: queued discovered runs now seed host-global occupancy in both poll refresh and direct admission, released queued drains remain non-occupying, and manifestless inflight claims resolve the queued run identity before synthetic claim keys so they are not double-counted. Added regressions for same-pipeline retained queued, foreign retained queued, direct foreign queued admission, released queued drains, and manifestless inflight queued runs.

File: orchestrator/src/cli/control/providerIssueHandoff.ts [hunk 4]
```diff
@@ -1487,6 +1491,19 @@ export function createProviderIssueHandoffService(
...
+      const claim = claimByProviderKey.get(providerKey) ?? null;
+      if (claim?.state === 'released') {
+        continue;
+      }
+      const occupancyKey = run.manifestPath || run.runId;
+      if (seededOccupancyKeys.has(occupancyKey)) {
+        continue;
+      }
+      seededOccupancyKeys.add(occupancyKey);
+      gate.noteOccupied({ state: claimStateByProviderKey.get(providerKey) ?? null });
```
[the remark this answers is not in the document]
Comment (Kbediako) [reply orchestrator/src/cli/control/providerIssueHandoff.ts]: Fixed in 455cf820f: direct admission now marks queued discovered occupancy with the retained claim's issue_state instead of the active-claim-only map, so queued In Progress retained work no longer counts as unknown against unrelated Ready per-state caps. Added regression coverage in the direct queued-worker test for a queued retained In Progress claim plus Ready state cap.

**Comment by Kbediako** [comment]
@codex review

**Comment by Kbediako** [comment]
@coderabbitai review

Current head: fd3c6d0b2a09fce8d5e501ac9b373aeb55fd387d. The released-claim identity concern was addressed in fd3c6d0b2, all review threads are resolved, CI is green, and the PR body/workpad are refreshed. Please re-review this head so the stale CHANGES_REQUESTED decision can clear if no actionable blockers remain.

**Comment by Kbediako** [comment]
@coderabbitai resume

Reviews are paused on this branch. Please resume and review current head fd3c6d0b2a09fce8d5e501ac9b373aeb55fd387d; all prior actionable threads are resolved and CI is green.

**Comment by Kbediako** [comment]
@codex review

Current head: ebe0329327d3be1c5640a9a5ba7dd3c72cc17855. CodeRabbit is green, required checks are green, and the latest checklist-only follow-up records the standalone-review startup-anchor boundary plus manual fallback. Please review this current head.
